src/test/java/cart/service/product/ProductServiceTest.java

```diff
@@ -0,0 +1,84 @@
+package cart.service.product;
+
+import cart.domain.product.Product;
+import cart.infrastructure.ProductDao;
+import cart.web.dto.ProductResponseDto;
+import cart.web.dto.ProductSaveRequestDto;
+import cart.web.dto.ProductUpdateRequestDto;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.jdbc.Sql;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@Sql(scripts = "classpath:schema.sql")
+@SpringBootTest
+class ProductServiceTest {
+
+    @Autowired
+    private ProductDao productDao;
+    @Autowired
+    private ProductService productService;
+
+    private Product insertProduct(String name, String imageUrl, int price) {
+        Product givenProduct = Product.builder()
+                .name(name)
+                .imageUrl(imageUrl)
+                .price(price)
+                .build();
+
+        return productDao.insert(givenProduct);
+    }
+
+    @Test
+    void findAll() {
+        insertProduct("상품A", "image.com/imageA", 10000);
+        insertProduct("상품B", "image.com/imageB", 20000);
+
+        List<ProductResponseDto> productResponseDtos = productService.findAll();
+        assertThat(productResponseDtos).flatExtracting(ProductResponseDto::getName).containsExactly("상품A", "상품B");
+        assertThat(productResponseDtos).flatExtracting(ProductResponseDto::getImageUrl)
+                .containsExactly("image.com/imageA", "image.com/imageB");
+        assertThat(productResponseDtos).flatExtracting(ProductResponseDto::getPrice).containsExactly(10000, 20000);
```

Comment:
```
테스트를 위한 값 준비

테스트 대상 실행

검증 실행
```

테스트를 세 부분으로 문단을 나눠주면 가독성을 증가시킬 수 있어요!

+) given when then 으로 나눠서 구분하는 방법도 있습니다 :)
```suggestion
        // given 
        insertProduct("상품A", "image.com/imageA", 10000);
        insertProduct("상품B", "image.com/imageB", 20000);

        // when
        List<ProductResponseDto> productResponseDtos = productService.findAll();
        
        // then
        assertThat(productResponseDtos).flatExtracting(ProductResponseDto::getName).containsExactly("상품A", "상품B");
        assertThat(productResponseDtos).flatExtracting(ProductResponseDto::getImageUrl)
                .containsExactly("image.com/imageA", "image.com/imageB");
        assertThat(productResponseDtos).flatExtracting(ProductResponseDto::getPrice).containsExactly(10000, 20000);
```

Reply:
넵 지금까지 작성할 때도 given when then을 염두에 두고 코드를 구현하긴 했지만 테스트 코드가 길어지니깐 확실히 가독성이 점점 떨어지는 것 같네요 🥲
given / when / then 주석을 달아주는 방식으로 모두 개선해보겠습니다~!!